jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c

```diff
@@ -2579,6 +2581,215 @@ ecma_builtin_array_prototype_object_copy_within (const ecma_value_t args[], /**<
+      if (depth > 0)
+      {
+        ecma_value_t is_array = ecma_is_value_array (element);
+
+        if (ECMA_IS_VALUE_ERROR (is_array))
+        {
+          ecma_free_value (element);
+          return is_array;
+        }
+        shouldFlatten = ecma_is_value_true (is_array);
+      }
+
+      if (shouldFlatten)
+      {
+        ecma_object_t *element_obj = ecma_get_object_from_value (element);
+        uint32_t elementLen;
+        ecma_value_t len_value = ecma_op_object_get_length (element_obj, &elementLen);
+
+        if (ECMA_IS_VALUE_ERROR (len_value))
+        {
+          ecma_deref_object (element_obj);
+          return len_value;
+        }
+
+        ecma_value_t targetIndex_val = ecma_builtin_array_flatten_into_array (target,
+                                                                              element_obj,
+                                                                              elementLen,
+                                                                              targetIndex,
+                                                                              depth-1,
+                                                                              ECMA_VALUE_UNDEFINED,
+                                                                              ECMA_VALUE_UNDEFINED);
+
+        ecma_deref_object (element_obj);
+
+        if (ECMA_IS_VALUE_ERROR (targetIndex_val))
+        {
+          return targetIndex_val;
+        }
+        targetIndex = (uint32_t) ecma_get_number_from_value (targetIndex_val);
+      }
+      else
+      {
+        uint8_t propertys = ECMA_PROPERTY_CONFIGURABLE_ENUMERABLE_WRITABLE;
+        ecma_value_t element_temp = ecma_builtin_helper_def_prop_by_index (ecma_get_object_from_value (target),
+                                                                           targetIndex,
+                                                                           element,
+                                                                           propertys);
+
+        ecma_free_value (element);
+        if (ECMA_IS_VALUE_ERROR (element_temp))
+        {
+          return element_temp;
+        }
+
+        targetIndex++;
+      }
+    }
+    sourceIndex++;
+  }
+  return ecma_make_uint32_value (targetIndex);
+} /* ecma_builtin_array_flatten_into_array */
+
+/**
+ * The Array.prototype object's 'flat' routine
+ *
+ * See also:
+ *          ECMA-262 v10, 22.1.3.10
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_builtin_array_prototype_object_flat (const ecma_value_t args[], /**< arguments list */
+                                          ecma_length_t args_number, /**< number of arguments */
+                                          ecma_object_t *obj_p, /**< array object */
+                                          uint32_t len) /**< array object's length */
+{
+  ecma_number_t depthNum = 1;
```

Comment:
Please tidy up you code with a few newlines. See the [coding guidlines](https://jerryscript.net/coding-standards/) for examples.

Reply:
Still camelcase variable name.